functions/createReview/astParsing/index.js

```diff
@@ -0,0 +1,101 @@
+import { Parser } from 'acorn'
+import tsPlugin from 'acorn-typescript'
+import path from 'node:path'
+import { LooseParser } from 'acorn-loose'
+import jsxPlugin from 'acorn-jsx'
+
+import validators from './rules/index.js'
+import { parseAndAggregate } from './parser.js'
+import { mapLineToDiff } from 'map-line-to-diff'
+
+function getParser(fileName) {
+  const fileExtension = path.extname(fileName)
+  let parser
+  if (fileExtension === '.ts' || fileExtension === '.tsx') {
+    parser = Parser.extend(tsPlugin())
+  } else if (fileExtension === '.jsx') {
+    parser = Parser.extend(jsxPlugin())
+  } else if (fileExtension === '.js') {
+    parser = Parser
+  }
+  return parser
+}
```

Comment:
nit: can you please turn it into a switch/case? it feels like it's a perfect fit for this logic as it won't require either mutable state nor equality comparisons

Reply:
Done. Will push change soon.